c7n/resources/sagemaker.py

```diff
@@ -66,14 +68,110 @@ class resource_type(object):
         service = 'sagemaker'
         enum_spec = ('list_training_jobs', 'TrainingJobSummaries', None)
         detail_spec = (
-            'describe_training_job', 'TrainingJobName',
-            'TrainingJobName', None)
+            'describe_training_job', 'TrainingJobName', 'TrainingJobName', None)
         id = 'TrainingJobArn'
         name = 'TrainingJobName'
         date = 'CreationTime'
         dimension = None
         filter_name = None
 
+    permissions = (
+        'sagemaker:ListTrainingJobs', 'sagemaker:DescribeTrainingJobs',
+        'sagemaker:ListTags')
+
+    def __init__(self, ctx, data):
+        super(SagemakerJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'InProgress'}]))
+
+    def resources(self, query=None):
+        q = self.resource_query()
+        if q is not None:
+            query = query or {}
+            for i in range(0, len(q)):
+                query[q[i]['Name']] = q[i]['Value']
+        return super(SagemakerJob, self).resources(query=query)
+
+    def resource_query(self):
+        result = []
```

Comment:
can this logic be moved to query parser.parse so that the base class can just do a pass through on the parsed outputs?

Reply:
I'll move it there and see what the results are.